src/router.rs

```diff
@@ -0,0 +1,72 @@
+use yew::{html::IntoPropValue, web_sys::Url};
+use yew_router::{components::RouterAnchor, prelude::*};
+
+#[derive(Clone, Debug, Switch)]
+pub enum AppRoute {
+    #[to = "/resources"]
+    Resources,
+    #[to = "/events"]
+    Events,
+    #[to = "/about"]
+    About,
+    #[to = "/!"]
+    Index,
+}
+impl AppRoute {
+    pub fn into_public(self) -> PublicUrlSwitch {
+        PublicUrlSwitch(self)
+    }
+
+    pub fn into_route(self) -> Route {
+        Route::from(self.into_public())
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct PublicUrlSwitch(AppRoute);
```

Comment:
Why do we need a Public URL Switch?

Reply:
We don't need the struct PublicUrlSwitch, I made a refactoring.